src/routing/core/mod.rs

```diff
@@ -82,31 +93,40 @@ impl Core {
             resource_proof: ResourceProof::new(RESOURCE_PROOF_DATA_SIZE, RESOURCE_PROOF_DIFFICULTY),
             end_users: EndUserRegistry::new(),
             connectivity_complaints: ConnectivityComplaints::new(),
+            lagging_messages: LaggingMessages::default(),
         }
     }
 
     ////////////////////////////////////////////////////////////////////////////
     // Miscellaneous
     ////////////////////////////////////////////////////////////////////////////
 
-    // Update our knowledge of their (sender's) section and their knowledge of our section.
-    pub(crate) fn update_section_knowledge(&mut self, msg: &Message) -> Result<Vec<Command>> {
+    fn check_for_entropy(
```

Comment:
what does the returning boolean flg mean?

Reply:
The boolean being returned is flag indicating whether we should be handling that message or not due to AE. So if  there is a mismatch with the keys in the message, we can sometimes act on them and in other scenarios not. So this boolean is used as a flag to indicate if we can act on the message.